### PR TITLE
rad-init: Adds name, description and branch to options

### PR DESCRIPTION
**Pull request by erak**

### CODE REVIEW

**Comment by geigerzaehler** [comment]
This looks good for our use-case in Upstream. Thanks!

File: init/src/lib.rs [hunk 1]
```diff
@@ -22,12 +22,19 @@ Usage
...
+    --help               Print help
 "#,
 };
 
+#[derive(Default)]
 pub struct Options {
     pub path: Option<PathBuf>,
+    name: Option<String>,
+    description: Option<String>,
+    branch: Option<String>,
```
Comment (cloudhead) [inline init/src/lib.rs:37]: These can all be public.

File: init/src/lib.rs [hunk 2]
```diff
@@ -37,8 +44,42 @@ impl Args for Options {
...
+                        .value()?
+                        .to_str()
+                        .ok_or(anyhow::anyhow!(
+                            "invalid project description specified with `--description`"
+                        ))?
+                        .to_owned();
+
+                    description = Some(value);
+                }
+                Long("branch") if branch.is_none() => {
```
Comment (cloudhead) [inline init/src/lib.rs:74]: Should be `"default_branch"` like in the usage instructions

File: init/src/lib.rs [hunk 3]
```diff
@@ -37,8 +44,42 @@ impl Args for Options {
...
+                    let value = parser
+                        .value()?
+                        .to_str()
+                        .ok_or(anyhow::anyhow!("invalid branch specified with `--branch`"))?
```
Comment (cloudhead) [inline init/src/lib.rs:78]: Same here

**Review by cloudhead — DISMISSED** [review]

**Comment by jgresham** [comment]
awesome, thank you!

**Pull request merged**